src/Microsoft.Azure.SignalR.Common/ClientInvocation/RoutedClientResultsManager.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using System.Threading;
+using Microsoft.AspNetCore.SignalR.Protocol;
+using System.Collections.Generic;
+using Microsoft.Azure.SignalR.Protocol;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal class RoutedClientResultsManager: IRoutedClientResultsManager
+    {
+        private readonly ConcurrentDictionary<string,  RoutedInvocation> _routedInvocations = new();
+        private readonly ConcurrentDictionary<string, List<string>> _serviceMappingMessages = new();
+
+        public void AddRoutedInvocation(string connectionId, string invocationId, string callerServerId, string instanceId, CancellationToken cancellationToken)
+        {
+            var cts = new CancellationTokenSource();
```

Comment:
why not directly register into cancellationToken `TryCompleteResult?

Reply:
fixed